[FEATURE] Add validator for date of birth

eliashaeussler opened the pull request:
This PR adds a new `DateIntervalValidator` and configures the `dateOfBirth` property with an interval of `P18Y` (18 years, configurable via site settings). If the member is younger than 18 years, an error is shown:

<img width="617" alt="image" src="https://github.com/user-attachments/assets/bfaf5020-53dc-42d6-a29d-1ca37a8ad625" />

Resolves: #11

eliashaeussler committed:
[FEATURE] Add validator for date of birth (must be >= 18 years)

Resolves: #11

boehmmatthias commented:
Wouldn't it be better if we make the minimum age to enter as a new member editable with a setting? As far as I know is this not necessarily 18 years. At least in Austria you are able to join without the consent of a guardian if you are at least 14 years old.

eliashaeussler committed:
Merge branch 'main' into feature/age-validator

eliashaeussler committed:
[FEATURE] Make minimum age of new members configurable

Resolves: #11

eliashaeussler commented:
> Wouldn't it be better if we make the minimum age to enter as a new member editable with a setting? As far as I know is this not necessarily 18 years. At least in Austria you are able to join without the consent of a guardian if you are at least 14 years old.

@boehmmatthias You're totally right. ;) I added a new site setting:

<img width="828" alt="image" src="https://github.com/user-attachments/assets/b8a27a6d-45eb-496c-aa1c-36bc353b284b" />

Unfortunately, the current implementation does not work due to a bug in Extbase. I added a patch to resolve this issue (https://review.typo3.org/c/Packages/TYPO3.CMS/+/89444). Until the patch is merged, this PR remains pending.

boehmmatthias commented on Configuration/Sets/MemberManagement/config.yaml:
```diff
@@ -21,3 +21,7 @@ settings:
         sepaCreditorId: ''
         paymentReason: ''
         paymentDueMonth: ''
+        paymentReminderPeriod: ''
```
Why did you add paymentReminderPeriod and paymentGracePeriod in this PR?

eliashaeussler replied:
You're right, it should not be part of this PR. I'll extract it into a separate PR.

eliashaeussler replied:
=> #27

boehmmatthias commented on Configuration/Sets/MemberManagement/settings.definitions.yaml:
```diff
@@ -13,6 +13,9 @@ categories:
     parent: Memsy.organization
+  Memsy.membership:
+    label: 'Membership'
+    parent: Memsy
```
Same here - why the addition of this definition in this PR?

eliashaeussler replied:
I added a new category `memberhsip` for the `minimumAgeInYears` setting, because it did not fit in any of the other categories.

eliashaeussler committed:
[TASK] Remove out of scope changes

eliashaeussler committed:
Merge branch 'main' into feature/age-validator

eliashaeussler commented:
Once the next TYPO3 patch version is released, we need to bump minimum version requirements due to the [fixed bug in `AbstractGenericObjectValidator`](https://review.typo3.org/c/Packages/TYPO3.CMS/+/89444).

eliashaeussler committed:
[TASK] Require at least TYPO3 13.4.13
